.github/workflows/build-and-tag-ranger-image.yml

```diff
@@ -0,0 +1,190 @@
+          ./download-archives.sh none
+
+      - name: Generate image ID
+        id: meta
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81
+        with:
+          images: |
+            ghcr.io/${{ github.repository_owner }}/ranger
+          tags: |
+            type=match,pattern=ranger-(.*),value={{branch}},group=1
+          flavor: |
+            latest=false
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@49b3bc8e6bdd4a60e6116a5414239cba5943d3cf
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@c47758b77c9736f4b2ef4073d4d51994fabfe349
+
+      - name: Login to GitHub Container Registry
+        id: login
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push images
+        id: build
+        working-directory: dev-support/ranger-docker
+        run: |
+          set -o allexport
+          source .env
+          set +o allexport
+          docker buildx build \
+            --build-arg RANGER_BASE_IMAGE=${RANGER_BASE_IMAGE} \
+            --build-arg RANGER_BASE_VERSION=${RANGER_BASE_VERSION} \
+            --build-arg RANGER_VERSION=${RANGER_VERSION} \
+            --build-arg RANGER_DB_TYPE=postgres \
+            --file Dockerfile.ranger \
+            --platform linux/amd64,linux/arm64 \
+            --tag ghcr.io/${{ github.repository_owner }}/ranger:${RANGER_VERSION} \
+            --push .
+          docker buildx build \
+            --build-arg POSTGRES_VERSION=${POSTGRES_VERSION} \
+            --file Dockerfile.ranger-postgres \
+            --platform linux/amd64,linux/arm64 \
+            --tag ghcr.io/${{ github.repository_owner }}/ranger-db:${RANGER_VERSION} \
+            --push .
+          docker buildx build \
+            --build-arg SOLR_VERSION=${SOLR_VERSION} \
+            --file Dockerfile.ranger-solr \
+            --platform linux/amd64,linux/arm64 \
+            --tag ghcr.io/${{ github.repository_owner }}/ranger-solr:${RANGER_VERSION} \
+            --push .
+          docker buildx build \
+            --build-arg ZK_VERSION=${ZK_VERSION} \
+            --file Dockerfile.ranger-zk \
+            --platform linux/amd64,linux/arm64 \
+            --tag ghcr.io/${{ github.repository_owner }}/ranger-zk:${RANGER_VERSION} \
+            --push .
+
+  tag:
+    needs: build
+    if: ${{ github.event.inputs.token != '' }}
+    runs-on: ubuntu-latest
+    env:
+      DOCKERHUB_USER: ${{ secrets.DOCKERHUB_USER }}
+      IMAGE_ID: ${{ needs.build.outputs.image-id }}
+      REGISTRIES: ghcr.io # docker.io is appended dynamically
+    steps:
+      - name: Verify OAuth Token
+        run: |
+          response=$(curl https://oauth.apache.org/token\?code\=${{ github.event.inputs.token }})
+          echo "$response" | jq -e . >/dev/null 2>&1
```

Comment:
- `curl` will output download progress to std.out, `jq` will fail to parse it.  Use `-LSs`
- Token shown in log in plain text.